code/game/objects/structures/guncase.dm

```diff
@@ -60,36 +60,57 @@
 	if(iscyborg(user) || isalien(user))
 		return
 	if(contents.len && open)
-		ShowWindow(user)
+		show_menu(user)
 	else
 		open = !open
 		update_icon()
 
-/obj/structure/guncase/proc/ShowWindow(mob/user)
-	var/dat = {"<div class='block'>
-				<h3>Stored Guns</h3>
-				<table align='center'>"}
-	if(LAZYLEN(contents))
-		for(var/i in 1 to contents.len)
-			var/obj/item/I = contents[i]
-			dat += "<tr><A href='?src=[REF(src)];retrieve=[REF(I)]'>[I.name]</A><br>"
-	dat += "</table></div>"
+/**
+  * show_menu: Shows a radial menu to a user consisting of an available weaponry for taking
+  *
+  * Arguments:
+  * * user The mob to which we are showing the radial menu
+  */
+/obj/structure/guncase/proc/show_menu(mob/user)
+	if(!LAZYLEN(contents))
+		return
 
-	var/datum/browser/popup = new(user, "gunlocker", "<div align='center'>[name]</div>", 350, 300)
-	popup.set_content(dat)
-	popup.open(FALSE)
+	var/list/display_names = list()
+	var/list/items = list()
+	for(var/i in 1 to contents.len)
+		var/obj/item/I = contents[i]
+		display_names[I.name + " ([i])"] = REF(I)
+		var/image/item_image = image(icon = I.icon, icon_state = I.icon_state)
+		if(length(I.overlays))
+			item_image.copy_overlays(I)
+		items += list(I.name + " ([i])" = item_image)
```

Comment:
```suggestion
	for(var/i in 1 to length(contents))
		var/obj/item/thing = contents[i]
		display_names[thing.name + " ([thing])"] = REF(thing)
		var/image/item_image = image(icon = thing.icon, icon_state = thing.icon_state)
		if(length(thing.overlays))
			item_image.copy_overlays(thing)
		items += list(thing.name + " ([thing])" = item_image)
```
Avoid single upper-case letters for variables, please.

Reply:
Fixed.